lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -2099,6 +2099,13 @@ func (d *zfs) MigrateVolume(vol Volume, conn io.ReadWriteCloser, volSrcArgs *mig
 		}
 	}
 
+	// If we haven't negotiated zvol support, ensure volume is not a zvol.
+	if !shared.StringInSlice(migration.ZFSFeatureZvolFilesystems, volSrcArgs.MigrationType.Features) {
```

Comment:
@monstermunchkin @stgraber should we fallback to doing rsync migration here?
As although there is the potential to lose some of the filesystem functionality on the target, this wouldn't be any different to, say, migrating from LVM to ZFS (which we currently allow).

Reply:
@stgraber confirmed we should not fallback and this error should remain.